src/winter_onboarding_2021/fleet_management_service/db/fleet.clj

```diff
@@ -1,5 +1,40 @@
 (ns winter-onboarding-2021.fleet-management-service.db.fleet
-  (:require [winter-onboarding-2021.fleet-management-service.db.core :as db]))
+  (:require [winter-onboarding-2021.fleet-management-service.db.core :as db]
+            [honey.sql :as sql]
+            [honey.sql.helpers :refer [select from limit offset join order-by where]]))
 
 (defn create [fleet]
   (db/insert! :fleets fleet))
+
+(defn total []
+  (db/query! (sql/format (-> (select [:%count.* :count])
+                             (from :fleets)))))
+
+(defn select-fleets-for-admin [admin-id off lim]
+  (db/query! (sql/format (-> (select :id
+                                     :name
+                                     :created-by
+                                     :created-at)
+                             (from :fleets)
+                             (join :user_fleet [:= :user_fleet.fleet_id :fleets.id])
+                             (limit lim)
```

Comment:
Pagination appears to be common pattern for any query that's listing rows. Why not extract it into function that takes a query and returns a new  paginated query? 

```
(def paginate [query limit offset]
   ...)
```

Reply:
```
(defn paginate [query off lim]
  (-> query
      (offset off)
      (limit lim)))

(defn user-fleets [admin-id off lim]
  (db/query!
   (sql/format
    (paginate (-> (select :fleets.id
                          :fleets.name
                          :fleets.created-by
                          :fleets.created-at)
                  (from :fleets)
                  (join :users_fleets [:= :users_fleets.fleet_id :fleets.id])
                  (where [:= :users_fleets.user_id admin-id])
                  (order-by :created-at))
              off
              lim))))
```

So @alishamohanty & I came up with this abstraction but we think it is not adding any advantage plus it affects the code readability. Did you have some other way of abstraction in mind?